### PR TITLE
Add --needed config option

### PR DESCRIPTION
**Pull request by c0d3z3r0**
Add --needed config option

### CODE REVIEW

File: doc/yay.8 [hunk 1]
```diff
@@ -326,6 +326,14 @@ only GitHub packages are supported.
 .B \-\-nodevel
 Do not check for development packages updates during sysupgrade.
 
+.TP
+.B \-\-needed
+Pass --needed option to pacman by default.
```
Comment (AladW) [inline doc/yay.8]: What does "by default" mean here? Doesn't that only apply if you set it in the config file?

Comment (c0d3z3r0) [reply doc/yay.8]: Exactly. This is under "PERMANENT CONFIGURATION SETTINGS" so were indeed speaking about the config file

Comment (Morganamilo) [reply doc/yay.8]: They only become permanent when `--save` is used. Without that it will not be by default.

**Comment by Morganamilo** [comment]
I don't think I'm a fan of incorporating pacman's flags into yay's config. If needed should be a config option it should be in pacman.conf. Although I don't see much of a use case anyway. The only time I ever use needed tends to be in group installs.

Also I'm afraid all this PR does is make needed not work. you've changed it from a pacman flag to a yay flag. So instead of passing it to pacman, yay now consumes it when setting the config option. But you don't do anything with the config option, it just kind of exists there, doing nothing.

**Comment by c0d3z3r0** [comment]
Ouch. I just realized I always used --needed but completely misunderstood (YES, I read the man page ;) ) what it does. "group installs" was the missing hint... Thank you!

**Pull request closed without merging**